build.rs

```diff
@@ -0,0 +1,54 @@
+mod cli {
+    include!("src/cli.rs");
+}
+
+const LONG_ABOUT: &str =
+    "prr is a tool that brings mailing list style code reviews to Github PRs. This \
+means offline reviews and inline comments, more or less.
+
+To that end, prr introduces a new workflow for reviewing PRs:
+  1. Download the PR into a \"review file\" on your filesystem
+  2. Mark up the review file using your favorite text editor
+  3. Submit the review at your convenience
+
+For full documentation, please visit https://doc.dxuuu.xyz/prr/.";
+
+fn main() -> std::io::Result<()> {
+    if let Some(out_path) = std::env::var_os("GEN_DIR").or(std::env::var_os("OUT_DIR")) {
+        use clap::CommandFactory;
+        #[allow(unused_variables)]
+        let out_dir = std::path::PathBuf::from(out_path);
+        #[allow(unused_mut, unused_variables)]
+        let mut cmd = cli::Cli::command()
+            .author("Daniel Xu <dxu@apache.org>")
+            .about("Mailing list style code reviews for GitHub")
+            .long_about(LONG_ABOUT);
+
+        #[cfg(feature = "clap_mangen")]
+        {
+            let man_dir = std::path::Path::join(&out_dir, "man");
+            std::fs::create_dir_all(&man_dir)?;
+            clap_mangen::generate_to(cmd.clone(), &man_dir)?;
+        }
+
+        #[cfg(feature = "clap_complete")]
```

Comment:
These features should be listed in Cargo.toml, right?

Reply:
Optional dependencies [implicitly defines a feature of the same name](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies), so you can just use --features=clap_mangen rather than defining a new one. Up to you whether you want to explicitly define in Cargo.toml.

Reply:
Ah, didn't know that. Thanks for explaining!